### PR TITLE
[fix][io] add protobuf ByteString to pulsar-io jdbc core

### LINKED ISSUE #20247: [Bug] Pulsar IO JDBC Clickhouse and Protobuf native with "ByteString" throws exception
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

Rocky Linux 8
Pulsar 2.11.1

### Minimal reproduce step

- Add ProtobufNative Schema to topic with "bytes" field.
- Use Pulsar IO JDBC Clickhouse Sink

### What did you expect to see?

Event is sinked/inserted to clickhouse.

### What did you see instead?

```
2023-05-07T13:06:54,039+0200 [pool-2-thread-1] ERROR org.apache.pulsar.io.jdbc.JdbcAbstractSink - Update count 0 not match total number of records 1005
2023-05-07T13:06:54,098+0200 [pool-2-thread-1] ERROR org.apache.pulsar.io.jdbc.JdbcAbstractSink - Got exception 
java.lang.Exception: Not support value type, need to add it. class com.google.protobuf.ByteString$LiteralByteString
        at org.apache.pulsar.io.jdbc.BaseJdbcAutoSchemaSink.setColumnValue(BaseJdbcAutoSchemaSink.java:186) ~[pulsar-io-jdbc-core-2.11.1.jar:?]
        at org.apache.pulsar.io.jdbc.BaseJdbcAutoSchemaSink.bindValue(BaseJdbcAutoSchemaSink.java:81) ~[pulsar-io-jdbc-core-2.11.1.jar:?]
        at org.apache.pulsar.io.jdbc.JdbcAbstractSink.flush(JdbcAbstractSink.java:247) ~[pulsar-io-jdbc-core-2.11.1.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

### Anything else?

As i'm aware of, clickhouse supports "ByteString": https://clickhouse.com/docs/en/sql-reference/data-types/fixedstring

Maybe adding `.setBytes` does the job, but i'm not aware how to build the pulsar io `.nar` files to test it. How do I build the connectors?

https://github.com/apache/pulsar/blob/master/pulsar-io/jdbc/core/src/main/java/org/apache/pulsar/io/jdbc/BaseJdbcAutoSchemaSink.java#L170
```
        } else if (value instanceof ByteString) {
            statement.setBytes(index, ((ByteString) value)).toByteArray();
```

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by bpereto**
- tested with jdbc clickhouse on pulsar 2.11.1

Fixes #20247

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [x] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-io/jdbc/core/pom.xml [hunk 1]
```diff
@@ -58,6 +58,13 @@
       <groupId>com.fasterxml.jackson.dataformat</groupId>
       <artifactId>jackson-dataformat-yaml</artifactId>
     </dependency>
+
+    <dependency>
+      <groupId>com.google.protobuf</groupId>
+      <artifactId>protobuf-java</artifactId>
+      <version>3.22.4</version>
```
Comment (tisonkun) [inline pulsar-io/jdbc/core/pom.xml:65]: Let's use the version in `dependencyManagement` in parent pom.

Also, cloud we use `provided` scope? (N.B. I don't take a considerate thought).

Comment (nicoloboschi) [reply pulsar-io/jdbc/core/pom.xml:65]: +1 we need to use `provided` to avoid runtime clashes
Also get the already declared version

Comment (bpereto) [reply pulsar-io/jdbc/core/pom.xml]: I set the version to `${protobuf3.version}` from `pulsar/pom.xml` and added the provided scope.
Is this what you ment by `dependencyManagement` ?

Comment (tisonkun) [reply pulsar-io/jdbc/core/pom.xml]: @bpereto Just remove the `<version>` line. We already import:

```xml
      <dependency>
        <groupId>com.google.protobuf</groupId>
        <artifactId>protobuf-bom</artifactId>
        <version>${protobuf3.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
```

Comment (tisonkun) [reply pulsar-io/jdbc/core/pom.xml]: @bpereto Also, please verify that `provided` statisfy your use case. One downside of `provided` is that - I'm not sure - if we don't pack protobuf into the NARball, it's possible to meet `NoClassDefFoundError`.

File: pulsar-io/jdbc/core/pom.xml [hunk 2]
```diff
@@ -58,6 +58,14 @@
       <groupId>com.fasterxml.jackson.dataformat</groupId>
       <artifactId>jackson-dataformat-yaml</artifactId>
     </dependency>
+
+    <dependency>
+      <groupId>com.google.protobuf</groupId>
+      <artifactId>protobuf-java</artifactId>
+      <version>${protobuf3.version}</version>
```
Comment (tisonkun) [inline pulsar-io/jdbc/core/pom.xml:65]:
```suggestion
```

This can be a bit bikeshedding to keep consistent with other dependency patterns.

Comment (tisonkun) [reply pulsar-io/jdbc/core/pom.xml]: I've pushed a commit for you - to apply suggestions and fix style violation.

**Review by tisonkun — APPROVED** [review]

**Review by nicoloboschi — APPROVED** [review]
👍

**Comment by bpereto** [comment]
Perfect, currently the sink is starting, need to test further with data that will be sinked.

File: pulsar-io/jdbc/core/pom.xml (hunk 1 above)
Comment (bpereto) [reply pulsar-io/jdbc/core/pom.xml]: I will test it with data, currently the sink is starting with the proposed changes.

**Comment by bpereto** [comment]
works as inteded for me.

**Comment by tisonkun** [comment]
Thanks for your verification! Merging...

**Comment by tisonkun** [comment]
BTW, I notice another PR of yours https://github.com/apache/pulsar/pull/14584 that we may make progress if you review the comments and rebase on master.

**Pull request merged**